worlds/tracker/docs/apworld-integration.md

```diff
@@ -0,0 +1,98 @@
+
+```python
+    def create_regions(self) -> None:
+        is_ut = getattr(self.multiworld, "generation_is_fake", False)
+        if self.randomly_rolled_property == 1 or is_ut:
+            # This location will always be created by UT regardless of what UT happens to roll
+            self.create_location(...)
+```
+
+## Generating without a YAML
+
+You can also make it so a YAML is not required to generate for your world. To do so, you must first store all options that affect generation in your slot data.
+
+```python
+    def fill_slot_data(self) -> dict[str, Any]:
+        # Store all defined options
+        option_fields = [field.name for field in dataclasses.fields(self.options)]
+        return {
+            # Keep other randomized results as above
+            "options": self.options.as_dict(*option_fields),
+        }
+```
+
+To make implementation simpler, instead of the large amount of adjustments you'd likely have to make in `interpret_slot_data`, you can instead turn it into a static method that simply returns the slot data. This will inform UT that instead of doing an initial generation when it opens, it will do a regeneration once connected to the multiworld.
+
+```python
+    @staticmethod
+    def interpret_slot_data(slot_data: dict[str, Any]) -> dict[str, Any]:
+        # Trigger a regen in UT
+        return slot_data
+```
+
+During the regeneration, you can access the slot data via `re_gen_passthrough` on the multiworld object. Early in the generation you can inspect this object to set any instance attributes or options needed from the slot data so your later functions will behave as expected.
+
+```python
+    def generate_early(self) -> None:
+        re_gen_passthrough = getattr(self.multiworld, "re_gen_passthrough", {})
+        if re_gen_passthrough and self.game in re_gen_passthrough:
+            # Get the passed through slot data from the real generation
+            slot_data: dict[str, Any] = re_gen_passthrough[self.game]
+
+            slot_options: dict[str, Any] = slot_data.get("options", {})
+            # Set all your options here instead of getting them from the yaml
+            for key, value in slot_options.items():
+                opt: Optional[Option] = getattr(self.options, key, None)
+                if opt is not None:
+                    # You can also set .value directly but that won't work if you have OptionSets
+                    setattr(self.options, key, opt.from_any(value))
+```
+
+The last thing to do is inform UT that your world can generate without a YAML.
```

Comment:
```suggestion
The last thing you can do is inform UT that your world can generate without a YAML. This is not necessary to do, but it may be convenient. It does mean that someone cannot edit their local YAML to see what would have been in logic had they chosen different options.
```